src/dbally/iql_generator/iql_generator.py

```diff
@@ -15,57 +20,151 @@
 from dbally.prompt.template import PromptTemplate
 from dbally.views.exposed_functions import ExposedFunction
 
-ERROR_MESSAGE = "Unfortunately, generated IQL is not valid. Please try again, \
-    generation of correct IQL is very important. Below you have errors generated by the system:\n{error}"
+IQLQueryT = TypeVar("IQLQueryT", bound=IQLQuery)
 
 
-class IQLGenerator:
+@dataclass
+class IQLGeneratorState:
+    """
+    State of the IQL generator.
     """
-    Class used to generate IQL from natural language question.
 
-    In db-ally, LLM uses IQL (Intermediate Query Language) to express complex queries in a simplified way.
-    The class used to generate IQL from natural language query is `IQLGenerator`.
+    filters: Optional[Union[IQLFiltersQuery, Exception]] = None
+    aggregation: Optional[Union[IQLAggregationQuery, Exception]] = None
+
+    @property
+    def failed(self) -> bool:
+        """
+        Checks if the generation failed.
 
-    IQL generation is done using the method `self.generate_iql`.
-    It uses LLM to generate text-based responses, passing in the prompt template, formatted filters, and user question.
+        Returns:
+            True if the generation failed, False otherwise.
+        """
+        return isinstance(self.filters, Exception) or isinstance(self.aggregation, Exception)
+
+
+class IQLGenerator:
+    """
+    Program that orchestrates all IQL operations for the given question.
```

Comment:
nit(non-blocking): this doesn't conform to my conception of what "program" is - for me "program" is a standalone piece of software that can be run by itself. I would just call this "class" or would loose the noun completely.

Reply:
That's fair, I can change it